Foundation/String/UTF8.hs

```diff
@@ -820,6 +822,76 @@ reverse s@(String ba) = runST $ do
                 _  -> return () -- impossible
             loop ms (si `offsetPlusE` nb) d
 
+-- Finds how many times we need to perform the replacement, and where are
+-- the insertion points.
+indices :: String -> String -> (Int, [Offset8])
```

Comment:
The type of this is a big weird. I'd have expected indices, and not the number of them.